src/main/kotlin/mate/academy/HelloWorldAnonymousClasses.kt

```diff
@@ -8,6 +8,35 @@ class HelloWorldAnonymousClasses {
     }
 
     fun sayHello(names: List<String>): List<String> {
-        // TODO: implement
+        val englishGreeting = object : HelloWorldGreeting {
+            override fun greet() = "Hello world"
+
+            override fun greetSomeone(someone: String) = "Hello $someone"
+        }
+
+        val frenchGreeting = object : HelloWorldGreeting {
+            override fun greet() = "Salut tout le monde"
+
+            override fun greetSomeone(someone: String) = "Salut $someone"
+        }
+
+        val spanishGreeting = object : HelloWorldGreeting {
+            override fun greet() = "Hola, mundo"
+
+            override fun greetSomeone(someone: String) = "Hola, $someone"
+        }
+
+        val greetingList = listOf(englishGreeting, frenchGreeting, spanishGreeting)
+
+        val generalGreetings = greetingList.map { it.greet() }
+
+        val personalGreetings = names
```

Comment:
You may add here some optimization for case if names list is empty

Reply:
Done.